core-java-security/src/main/java/com/baeldung/ssl/EnableTLSv12.java

```diff
@@ -0,0 +1,134 @@
+package com.baeldung.ssl;
+
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.io.OutputStreamWriter;
+import java.io.PrintWriter;
+import java.net.URL;
+import java.security.SecureRandom;
+
+import javax.net.ssl.HttpsURLConnection;
+import javax.net.ssl.SSLContext;
+import javax.net.ssl.SSLParameters;
+import javax.net.ssl.SSLSocket;
+import javax.net.ssl.SSLSocketFactory;
+
+public class EnableTLSv12 {
+
+    public String url = "";
+    public Integer port = null;
+
+    public EnableTLSv12() {
+    }
+
+    public static void main(String[] args) {
+        try {
+            EnableTLSv12 enableTLSv12 = new EnableTLSv12();
+            if (args.length != 2) {
+                System.out.println("Provide the server url and the secure port:");
+                System.exit(-1);
+            }
+            enableTLSv12.getHost(args);
+            enableTLSv12.getPort(args);
+            enableTLSv12.enableTLSv12UsingHttpConnection();
+            enableTLSv12.enableTLSv12UsingProtocol();
+            enableTLSv12.enableTLSv12UsingSSLContext();
+            enableTLSv12.enableTLSv12UsingSSLParameters();
+        } catch (Exception e) {
+            e.printStackTrace();
```

Comment:
If we just want to print the stack trace, then go ahead and change the `main` method signature to throw the appropriate exceptions.